service/src/test/java/bio/terra/workspace/app/configuration/external/controller/FolderApiControllerTest.java

```diff
@@ -0,0 +1,545 @@
+        .description(description)
+        .displayName(displayName)
+        .parentFolderId(parentFolderId);
+  }
+
+  private ApiFolder updateFolder(
+      UUID workspaceId,
+      UUID folderId,
+      @Nullable String newDisplayName,
+      @Nullable String newDescription,
+      @Nullable UUID parentFolderId,
+      boolean updateParent)
+      throws Exception {
+    String serializedUpdateResponse =
+        updateFolderExpectCode(
+                workspaceId,
+                folderId,
+                newDisplayName,
+                newDescription,
+                parentFolderId,
+                updateParent,
+                HttpStatus.SC_OK)
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    return objectMapper.readValue(serializedUpdateResponse, ApiFolder.class);
+  }
+
+  private ResultActions updateFolderExpectCode(
+      UUID workspaceId,
+      UUID folderId,
+      String newDisplayName,
+      String newDescription,
+      @Nullable UUID parentFolderId,
+      boolean updateParent,
+      int code)
+      throws Exception {
+    return mockMvc
+        .perform(
+            addAuth(
+                patch(String.format(FOLDER_V1_PATH_FORMAT, workspaceId, folderId))
+                    .contentType(MediaType.APPLICATION_JSON_VALUE)
+                    .accept(MediaType.APPLICATION_JSON)
+                    .characterEncoding("UTF-8")
+                    .content(
+                        getUpdateRequestInJson(
+                            newDisplayName, newDescription, parentFolderId, updateParent)),
+                USER_REQUEST))
+        .andExpect(status().is(code));
+  }
+
+  private String getUpdateRequestInJson(
+      @Nullable String newDisplayName,
+      @Nullable String newDescription,
+      @Nullable UUID parentFolderId,
+      boolean updateParent)
+      throws JsonProcessingException {
+    var requestBody =
+        new ApiUpdateFolderRequestBody()
+            .description(newDescription)
+            .displayName(newDisplayName)
+            .parentFolderId(parentFolderId)
+            .updateParent(updateParent);
+    return objectMapper.writeValueAsString(requestBody);
+  }
+
+  private ApiFolder getFolder(UUID workspaceId, UUID folderId) throws Exception {
+    String folderGetResponse =
+        getFolderExpectCode(workspaceId, folderId, HttpStatus.SC_OK)
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+    return objectMapper.readValue(folderGetResponse, ApiFolder.class);
+  }
+
+  private ResultActions getFolderExpectCode(UUID workspaceId, UUID folderId, int code)
+      throws Exception {
+    return mockMvc
+        .perform(
+            addAuth(get(String.format(FOLDER_V1_PATH_FORMAT, workspaceId, folderId)), USER_REQUEST))
+        .andExpect(status().is(code));
+  }
+
+  private ApiFoldersList listFolders(UUID workspaceId) throws Exception {
+    String foldersGetResponse =
```

Comment:
This is list response, not get response. Rather than fixing, I suggest just calling everything serializedResponse. Then you don't have to edit for each method. (And people copy-pasting this won't have to edit for each method.)